lib/redispub/publisher.go

```diff
@@ -146,8 +152,9 @@ func publishSingleMessageWithRetries(p *Publication, maxRetries int, sleepTime t
 	return errors.Errorf("sending message (retried %v times)", maxRetries)
 }
 
-func publishSingleMessage(p *Publication, client redis.UniversalClient, prefix string, dedupeExpirationSeconds int) error {
+func publishSingleMessage(p *Publication, client redis.UniversalClient, prefix string, dedupeExpirationSeconds int, metricLastOplogEntryStaleness prometheus.Gauge) error {
 	start := time.Now()
+	metricLastOplogEntryStaleness.Set(float64(time.Since(time.Unix(int64(p.OplogTimestamp.T), 0))))
```

Comment:
I think you need to change this to `float64(time.Since(time.Unix(int64(p.OplogTimestamp.T), 0)).Seconds())` to have it be in seconds